File: examples/node/remote_gateway_embed.mjs
import {
  CogentClient,
  addGatewayRemote,
  printEmbedding,
  readRemoteArgs,
} from './_common.mjs';

const { alias, input } = readRemoteArgs('CogentClient remote embedding smoke input.');
const client = new CogentClient();
const endpoint = addGatewayRemote(client, alias);
const result = await client.embed({
  endpoint,
  input,
}).response;
printEmbedding(result);

File: examples/rust/src/bin/remote_gateway_embed.rs
use cogentlm::{CogentClient, CogentEmbedRequest};
use cogentlm_rust_examples::remote_common;
use futures::executor::block_on;

fn main() -> remote_common::ExampleResult<()> {
    block_on(async {
        let args = remote_common::args("CogentClient remote embedding smoke input.")?;
        let mut client = CogentClient::new();
        let endpoint = client.add_remote(
            args.alias.clone(),
            remote_common::gateway_remote(args.alias)?,
        )?;
        let response = client
            .embed(CogentEmbedRequest {
                endpoint: Some(endpoint),
                input: args.input,
                ..Default::default()
            })
            .await?;
        remote_common::print_embedding(response);
        Ok(())
    })
}
